code/modules/mob/living/carbon/xenomorph/castes/puppeteer/abilities_puppeteer.dm

```diff
@@ -0,0 +1,466 @@
+		return FALSE
+	owner_xeno.visible_message(span_warning("[owner_xeno] forms a repulsive puppet!"))
+	succeed_activate()
+
+/datum/action/xeno_action/activable/puppet/use_ability(atom/target)
+	var/turf/target_turf = get_turf(target)
+
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	huskaction.add_puppet(new /mob/living/carbon/xenomorph/puppet(target_turf, owner))
+	add_cooldown()
+
+// ***************************************
+// *********** Organic Bomb
+// ***************************************
+/datum/action/xeno_action/activable/organic_bomb
+	name = "Organic Bomb"
+	action_icon_state = "organic_bomb"
+	desc = "Causes one of our puppets to detonate on selection, spewing acid out of the puppet's body in all directions, gibbing the puppet."
+	cooldown_timer = 30 SECONDS
+	plasma_cost = 100
+	target_flags = XABB_MOB_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ORGANICBOMB,
+	)
+
+/datum/action/xeno_action/activable/organic_bomb/use_ability(mob/living/victim)
+	. = ..()
+	var/datum/action/xeno_action/activable/refurbish_husk/huskaction = owner.actions_by_path[/datum/action/xeno_action/activable/refurbish_husk]
+	if(length(huskaction.puppets) <= 0)
+		owner.balloon_alert(owner, "no puppets")
+		return fail_activate()
+	if(!istype(victim, /mob/living/carbon/xenomorph/puppet) || !(victim in huskaction.puppets))
+		victim.balloon_alert(owner, "not our puppet")
+		return fail_activate()
+	if(!SEND_SIGNAL(victim, COMSIG_PUPPET_CHANGE_ORDER, PUPPET_SEEK_CLOSEST))
+		victim.balloon_alert(owner, "fail")
+		return fail_activate()
+	RegisterSignal(victim, COMSIG_XENOMORPH_ATTACK_LIVING, PROC_REF(start_exploding))
+	RegisterSignal(victim, COMSIG_MOB_DEATH, PROC_REF(detonate))
+	addtimer(CALLBACK(src, PROC_REF(detonate), victim), 15 SECONDS)
+	add_cooldown()
+
+///asynchronous signal handler for start_exploding_async
+/datum/action/xeno_action/activable/organic_bomb/proc/start_exploding(mob/living/puppet)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, PROC_REF(start_exploding_async), puppet)
+
+///makes a puppet start a do_after to dexplode
+/datum/action/xeno_action/activable/organic_bomb/proc/start_exploding_async(mob/living/puppet)
+	puppet.visible_message(span_danger("[puppet] bloats and slowly unfurls its stitched body!"))
+	if(do_after(puppet, 1.5 SECONDS, FALSE, puppet, BUSY_ICON_DANGER))
+		detonate(puppet)
+
+///detonates a puppet causing a spray of acid
+/datum/action/xeno_action/activable/organic_bomb/proc/detonate(mob/living/puppet)
+	SIGNAL_HANDLER
+	UnregisterSignal(puppet, list(COMSIG_XENOMORPH_ATTACK_LIVING, COMSIG_MOB_DEATH))
+	if(QDELETED(puppet))
+		return
+	puppet.visible_message(span_danger("[puppet] ruptures, releasing corrosive acid!"))
+	var/turf/our_turf = get_turf(puppet)
+	playsound(our_turf, 'sound/bullets/acid_impact1.ogg', 50, 1)
+	puppet.gib()
+
+	for(var/turf/acid_tile AS in RANGE_TURFS(1, our_turf))
```

Comment:
this is pretty weak.

Reply:
weak no longer

Reply:
range is buffed to 2 which is already pretty big